Tests/NIOTests/ChannelTests.swift

```diff
@@ -1564,8 +1565,19 @@ public class ChannelTests: XCTestCase {
         class NoChannelReadVerificationHandler: ChannelInboundHandler {
             typealias InboundIn = ByteBuffer
 
+            private var seenEOF: Bool = false
+
+            public func userInboundEventTriggered(ctx: ChannelHandlerContext, event: Any) {
+                if case .some(ChannelEvent.inputClosed) = event as? ChannelEvent {
+                    self.seenEOF = true
+                }
+                ctx.fireUserInboundEventTriggered(event)
+            }
+
             public func channelRead(ctx: ChannelHandlerContext, data: NIOAny) {
-                XCTFail("Should not be called as autoRead is false and we did not call read(), but received \(self.unwrapInboundIn(data))")
+                if !self.seenEOF {
+                    XCTFail("Should not be called before seeing the EOF as autoRead is false and we did not call read(), but received \(self.unwrapInboundIn(data))")
```

Comment:
Is this right? Surely we expect to see the EOF *after* the read data, not before it.

Reply:
you're right :)